Library/Homebrew/keg_relocate.rb

```diff
@@ -163,6 +167,45 @@ def replace_text_in_files(relocation, files: nil)
     changed_files
   end
 
+  def replace_build_prefix_in_file(old_cellar, new_prefix, new_cellar, file)
+    old_prefix = Pathname(old_cellar).parent.to_s
+
+    old_prefix_optdir = "#{old_prefix}/opt"
+    old_prefix_etcdir = "#{old_prefix}/etc"
+    old_prefix_vardir = "#{old_prefix}/var"
+    old_prefix_vim_sharedir = "#{old_prefix}/share/vim"
+
+    new_prefix_optdir = "#{new_prefix}/opt"
+    new_prefix_etcdir = "#{new_prefix}/etc"
+    new_prefix_vardir = "#{new_prefix}/var"
+    new_prefix_vim_sharedir = "#{new_prefix}/share/vim"
+
+    old_prefix_dirs = [old_cellar, old_prefix_optdir, old_prefix_etcdir, old_prefix_vardir, old_prefix_vim_sharedir]
+    old_prefix_regex = Regexp.new(old_prefix_dirs.join("|"))
+
+    file.ensure_writable do
+      binary = File.binread file
+      # Skip patching if no string references to the cellar exist.
+      next if binary.scan(old_prefix_regex).empty?
```

Comment:
Do we do a similar scan in other relocations? It'd be nice to avoid rereading all ELF/MachO files multiple times if possible.

Reply:
See the below comment about storing this in `Tab`/`INSTALL_RECEIPT.json` during bottling.